scripts/check_icons.sh

```diff
@@ -2,44 +2,83 @@
     error=0
     file="$1"
-    geometry="$2"
-    content=$(identify -verbose "$file")
-
-    if [[ "$geometry" != "no-check" ]]; then
-        if ! echo "$content" | grep -q "Geometry: $geometry"; then
-            echo "ERROR => Icon should have a $geometry geometry"
-            error=1
-        fi
-    fi
+    device="$2"
+
+    case "$device" in
+        "nanos")
+            geometry="16x16"
+            ;;
+        "nanox")
+            geometry="14x14"
+            ;;
+        "nanos2")
+            geometry="14x14"
+            ;;
+        "stax")
+            geometry="32x32"
+            ;;
+        *)
+            echo "ERROR => Device '$device' not recognized"
+            return 1
+            ;;
+    esac
 
-    if echo "$content" | grep -q "Alpha"; then
-        echo "ERROR => Icon should have no alpha channel"
+    if ! identify -verbose "$file" | grep -q "Geometry: $geometry"; then
+        echo "ERROR => Icon '$file' for '$device' should have a '$geometry' geometry"
         error=1
+    else
+        echo "SUCCESS => Icon '$file' for '$device' has a '$geometry' geometry"
     fi
 
-    if ! echo "$content" | grep -q "Colors: 2"; then
-        echo "ERROR => Icon should have only 2 colors"
-        error=1
+    return "$error"
+)
+
+check_glyph() (
+    error=0
+    file="$1"
+    extension=$(basename "$file" | cut -d'.' -f2)
+    if [[ "$extension" != "gif" && "$extension" != "bmp" ]]; then
+        echo "ERROR => Glyph extension should be .gif or .bmp, not '.$extension'";
+        return 1
     fi
 
-    if ! echo "$content" | grep -q "0.*0.*0.*black"; then
-        echo "ERROR => Icon should have the black color defined"
+    content=$(identify -verbose "$file")
+
+    if echo "$content" | grep -q "Alpha"; then
+        echo "ERROR => Glyph should have no alpha channel"
         error=1
     fi
 
-    if ! echo "$content" | grep -q "255.*255.*255.*white"; then
-        echo "ERROR => Icon should have the white color defined"
-        error=1
+    if [[ "$extension" == "gif" ]]; then
+        if ! echo "$content" | grep -q "Colors: 2"; then
+            echo "ERROR => Glyph should have only 2 colors"
+            error=1
+        fi
+
+        if ! echo "$content" | grep -q "0.*0.*0.*black"; then
+            echo "ERROR => Glyph should have the black color defined"
+            error=1
+        fi
+
+        if ! echo "$content" | grep -q "255.*255.*255.*white"; then
+            echo "ERROR => Glyph should have the white color defined"
+            error=1
+        fi
+    else
+        if ! echo "$content" | grep -q "Depth: 1 bits-per-pixel component"; then
+            echo "ERROR => Glyph should 1 bit depth"
+            error=1
+        fi
     fi
 
     if [[ error -eq 0 ]]; then
-        echo "SUCCESS => Icon \"$file\" is compliant"
+        echo "SUCCESS => Glyph '$file' is compliant"
     else
         echo "FAILURE =>"
-        echo "identify -verbose \"$file\""
+        echo "identify -verbose '$file'"
```

Comment:
I think this is a bit to verbose. Maybe we can replace it with: "check identify -verbose ..."

Reply:
ok